int64 all
int64 upper
int64 lower
int64 outer
int64 shoes
---
#No return values